Further README Updates (#2028)
ethereumjs/ethereumjs-monorepo: packages/blockchain/README.md

## Introduction

> The `Blockchain` package represents an Ethereum-compatible blockchain storing a sequential chain of [@ethereumjs/block](../block) blocks and holding information about the current canonical head block as well as the context within the chain is operating (e.g. the hardfork rules the current head block adheres to).

Reviewer: ```suggestion
The `Blockchain` package represents an Ethereum-compatible blockchain storing a sequential chain of [@ethereumjs/block](../block) blocks and holding information about the current canonical head block as well as the context the chain is operating in (e.g. the hardfork rules the current head block adheres to).
```

> New blocks can be added to the blockchain and it is validated along if both the block format adheres to the given chain rules (with the `Blockchain.validateBlock()` function) as well as consensus rules apply (`Blockchain.consensus.validateConsensus()`).

Reviewer: ```suggestion
New blocks can be added to the blockchain. Validation ensures that the block format adheres to the given chain rules (with the `Blockchain.validateBlock()` function) and consensus rules (`Blockchain.consensus.validateConsensus()`).
```

> The library also supports reorg scenarios e.g. by allowing to add a new block with `Blockchain.putBlock()` which follows a different canoncial path to the head than given by the current canonical head block.

Reviewer: ```suggestion
The library also supports reorg scenarios e.g. by allowing to add a new block with `Blockchain.putBlock()` which follows a different canonical path to the head than given by the current canonical head block.
```

**WARNING**: Since `@ethereumjs/blockchain` is also doing write operations on the DB for safety reasons only run this on a copy of your database, otherwise this might lead to a compromised DB state.

## Consensus

Starting with v6 there is a dedicated consensus class for each type of supported consensus, `Ethash`, `Clique` and `Casper` (PoS, this one is rather the do-nothing part of `Casper` and letting the respective consensus/beacon client do the hard work! 🙂). Each consensus class adheres to a common interface `Consensus` implementing the following five methods in a consensus-specific way:

- `genesisInit(genesisBlock: Block): Promise<void>`
- `setup(): Promise<void>`
- `validateConsensus(block: Block): Promise<void>`
- `validateDifficulty(header: BlockHeader): Promise<void>`
- `newBlock(block: Block, commonAncestor?: BlockHeader, ancientHeaders?: BlockHeader[]): Promise<void>`

> For applying a modfied version of an existing consensus mechanism or applying a different mechanism an own consensus class can be written and passed in to the library with the `consensus` option along instantiation.

Author: Yeah that would be nice, the tests are actually in many cases my first go-to spot for inspiration or boilerplate for example code. 🙂 

Do you think taking over the whole (or reduced) thing as example code would be too extensive here in this case? Or would this also be an option?

Have also taken the occasion to reactivate our "Tiny Fixes Meta Issue" issue #2032 and added this task to the list.

## Custom Genesis State

Starting with v6 responsibility for setting up a custom genesis state moved from the [Common](../common/) library to the `Blockchain` package, see PR [#1924](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1924) for some work context.

> A genesis state can be set along `Blockchain` creation by passing in a custom `genesisBlock` and `genesisState`. For `mainnet` and the official test networks like `sepolia` or `goerli` genesis is already provided with the block data coming from `@ethereumjs/common`. The genesis state is being integrated in the `Blockchain` library (see `genesisStates` folder).

Reviewer 2: ```suggestion
A genesis state can be set along `Blockchain` creation by passing in a custom `genesisBlock` and `genesisState`. For `mainnet` and the official test networks like `sepolia` or `goerli` genesis is already provided with the block data coming from `@ethereumjs/common`. The genesis state has been integrated in the `Blockchain` library (see `genesisStates` folder).
```

> TODO: add code example here!

Reviewer: Just a note that there's a todo left here in case it's unintentional
